src/builder.rs

```diff
@@ -120,7 +130,185 @@ impl<'ctx> Builder<'ctx> {
 
         unsafe { InstructionValue::new(value) }
     }
+}
 
+#[llvm_versions(14.0..=latest)]
+impl<'ctx> Builder<'ctx> {
+    /// Builds a function call instruction.
+    /// [`FunctionValue`]s can be implicitly converted into a [`CallableValue`].
+    /// See [`CallableValue`] for details on calling a [`PointerValue`] that points to a function.
+    ///
+    /// [`FunctionValue`]: crate::values::FunctionValue
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// use inkwell::context::Context;
+    ///
+    /// // A simple function which calls itself:
+    /// let context = Context::create();
+    /// let module = context.create_module("ret");
+    /// let builder = context.create_builder();
+    /// let i32_type = context.i32_type();
+    /// let fn_type = i32_type.fn_type(&[i32_type.into()], false);
+    /// let fn_value = module.add_function("ret", fn_type, None);
+    /// let entry = context.append_basic_block(fn_value, "entry");
+    /// let i32_arg = fn_value.get_first_param().unwrap();
+    /// let md_string = context.metadata_string("a metadata");
+    ///
+    /// builder.position_at_end(entry);
+    ///
+    /// let ret_val = builder.build_call(fn_value, &[i32_arg.into(), md_string.into()], "call")
+    ///     .try_as_basic_value()
+    ///     .left()
+    ///     .unwrap();
+    ///
+    /// builder.build_return(Some(&ret_val));
+    /// ```
+    pub fn build_call<F>(&self, function: F, args: &[BasicMetadataValueEnum<'ctx>], name: &str) -> CallSiteValue<'ctx>
```

Comment:
Rather than duplicating the entire methods, could we just `llvm_versions`/`cfg` the new calls in the method itself?

Reply:
Actually I prefer that style too, but unfortunately compiler says `custom attributes cannot be applied to statements`...

Reply:
Ah, but we can do it with `#[cfg(not(feature="llvm14-0"))]`. I will fix them.